multiversx_sdk_core/transaction_builders/delegation_factory.py

```diff
@@ -0,0 +1,430 @@
+from multiversx_sdk_core.errors import ErrListsLengthDoNotMatch
+from multiversx_sdk_core.interfaces import (IAddress, IChainID, IGasLimit,
+                                            IGasPrice, INonce, ISignature,
+                                            ITransactionPayload,
+                                            ITransactionValue,
+                                            IValidatorPublicKey)
+from multiversx_sdk_core.serializer import arg_to_string
+from multiversx_sdk_core.transaction import Transaction
+
+
+class IConfig(Protocol):
+    chain_id: IChainID
+    min_gas_price: IGasPrice
+    min_gas_limit: IGasLimit
+    gas_limit_per_byte: IGasLimit
+
+
+class MetaChainSystemSCsCost:
+    STAKE = 5000000
+    UNSTAKE = 5000000
+    UNBOND = 5000000
+    CLAIM = 5000000
+    GET = 5000000
+    CHANGE_REWARD_ADDRESS = 5000000
+    CHANGE_VALIDATOR_KEYS = 5000000
+    UNJAIL = 5000000
+    DELEGATION_MANAGER_OPS = 50000000
+    DELEGATION_OPS = 1000000
+    UNSTAKE_TOKENS = 5000000
+    UNBOND_TOKENS = 5000000
+
+
+ADDITIONAL_GAS_LIMIT_PER_NODE = 6000000
+ADDITIONAL_GAS_FOR_OPERATIONS = 10000000
+
+
+class DelegationFactory:
+    def __init__(self, config: IConfig) -> None:
+        self.config = config
+
+    def create_new_delegation_contract(self,
+                                       sender: IAddress,
+                                       receiver: IAddress,
+                                       total_delegation_cap: int,
+                                       service_fee: int,
+                                       value: ITransactionValue,
+                                       transaction_nonce: Optional[INonce] = None,
+                                       gas_price: Optional[IGasPrice] = None,
+                                       gas_limit: Optional[IGasLimit] = None) -> Transaction:
+        data = self._prepare_data_for_create_new_delegation_contract(total_delegation_cap, service_fee)
+
+        transaction = self.create_transaction(
+            sender=sender,
+            receiver=receiver,
+            data_parts=data,
+            execution_gas_limit=MetaChainSystemSCsCost.DELEGATION_MANAGER_OPS + ADDITIONAL_GAS_FOR_OPERATIONS,
+            gas_limit_hint=gas_limit,
+            gas_price=gas_price,
+            nonce=transaction_nonce,
+            value=value
+        )
+
+        return transaction
+
+    def _prepare_data_for_create_new_delegation_contract(self, total_delegation_cap: int, service_fee: int) -> List[str]:
+        function = "createNewDelegationContract"
+        args_list: List[Any] = [
+            function,
+            arg_to_string(total_delegation_cap),
+            arg_to_string(service_fee)
+        ]
+        return args_list
+
+    def add_nodes(self,
+                  sender: IAddress,
+                  delegation_contract: IAddress,
+                  public_keys: Sequence[IValidatorPublicKey],
+                  signed_messages: List[ISignature],
```

Comment:
:+1:

Reply:
Can be both `Sequence` or both `List` (for consistency).

Reply:
Both are of type `Sequence` now.